### PR TITLE
docs(upgrade): improve upgrade procedure, data migration

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

File: en_US/changes/upgrade-4.3.md [hunk 1]
```diff
@@ -17,7 +17,27 @@ can be migrated with data export and import [command](#database-migration) runs.
...
+* `/opt/emqx/data` when running in docker (typically a mounted volume)
+* `<install-path>/data` when installed from direct zip package
+* `/var/lib/emqx/data` when installed from RPM or DEB packages
+
+### Copy Data directory
+
+When upgrading from 4.2 to 4.3 in a different host,
+the `data` directory should be copied from the 4.2 host to the new 4.3 host.
+
+However due to datbase schema changes, the old files in `data/mnesia` should not be copied,
```
Comment (thalesmg) [inline en_US/changes/upgrade-4.3.md]: Nit: _database_

File: en_US/changes/upgrade-4.3.md [hunk 2]
```diff
@@ -17,7 +17,27 @@ can be migrated with data export and import [command](#database-migration) runs.
...
+or deleted on the 4.3 node before starting EMQ X.
```
Comment (thalesmg) [inline en_US/changes/upgrade-4.3.md]: Just for a bit of clarification for myself: if the mnesia folder is inside `data` but it shouldn't be copied nor deleted, does this mean that it should be excluded from the destination host?

Something like the following?

```sh
cp -R old_host/data new_host/
rm -rf new_host/data/mnesia
```

Comment (zmstone) [reply en_US/changes/upgrade-4.3.md]: yes, that's right.

Comment (thalesmg) [reply en_US/changes/upgrade-4.3.md]: Ok, thanks!
In that case, I'd suggest changing the phrase a bit to help avoid my confusion:

> the old files in `data/mnesia` **should not** be copied ~or deleted~ to the 4.3 node before starting EMQ X.

**Review by thalesmg — APPROVED** [review]

**Pull request merged**